### ISSUE TITLE
Git Push does not work with Workflow dispatch

### ISSUE THREAD

**Issue by Legion2** [actions/checkout#451]
I try to push a commit in a manually triggered workflow. I basically use the "Push a commit using the built-in token" example as shown in the readme using a `workflow_dispatch` triggered workflow:
```yaml
      - uses: actions/checkout@v2
...      
      - name: Push updated files
        run: |
          git config user.name github-actions
          git config user.email github-actions@github.com
          git add .
          git commit -m "updated version of ${{ github.event.inputs.platform-dir }} to ${{ github.event.inputs.version }}"
          git push
```

When the workflow is triggered, the `git push` command failed with the following output:
```
 ! [rejected]        master -> master (fetch first)
error: failed to push some refs to 'https://github.com/Spegs21/CorsairLightingProtocolBoards'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

Do I have to configure something that it works with `workflow_dispatch`?
